shared/src/business/useCases/judgeActivityReport/getCasesByStatusAndByJudgeInteractor.ts

```diff
@@ -48,47 +65,43 @@ const filterCasesWithUnwantedDocketEntryEventCodes = caseRecords => {
   return caseRecordsToReturn;
 };
 
-/**
- * getCasesByStatusAndByJudgeInteractor
- * @param {object} applicationContext the application context
- * @param {object} providers the providers object
- * @param {string} providers.judgeName the name of the judge
- * @param {array} providers.statuses statuses of cases for judge activity
- * @returns {object} errors (null if no errors)
- */
 export const getCasesByStatusAndByJudgeInteractor = async (
   applicationContext,
-  {
-    judgeName,
-    statuses,
-  }: {
-    judgeName: string;
-    statuses: string[];
-  },
-) => {
+  params: JudgeActivityReportCavAndSubmittedCasesRequest,
+): Promise<{
+  cases: CavAndSubmittedFilteredCasesType[];
+  consolidatedCasesGroupCountMap: ConsolidatedCasesGroupCountMapResponseType;
+  totalCount: number;
+}> => {
   const authorizedUser = applicationContext.getCurrentUser();
 
   if (!isAuthorized(authorizedUser, ROLE_PERMISSIONS.JUDGE_ACTIVITY_REPORT)) {
     throw new UnauthorizedError('Unauthorized');
   }
 
-  const submittedAndCavCasesResults = await applicationContext
+  const searchEntity = new JudgeActivityReportSearch(params);
+  if (!searchEntity.isValid()) {
+    throw new InvalidRequest();
+  }
+
+  const { foundCases: submittedAndCavCasesResults } = await applicationContext
     .getPersistenceGateway()
     .getDocketNumbersByStatusAndByJudge({
       applicationContext,
-      judgeName,
-      statuses,
+      params: {
+        judges: searchEntity.judges,
+        statuses: searchEntity.statuses,
+      },
     });
 
   const rawCaseRecords: RawCase[] = await Promise.all(
-    submittedAndCavCasesResults.map(async result => {
-      return await applicationContext
-        .getPersistenceGateway()
-        .getCaseByDocketNumber({
+    submittedAndCavCasesResults.map(
```

Comment:
I think it would be possible to remove this call to fetch the full case if the `getDocketNumbersByStatusAndByJudge` query included `leadDocketNumber` as a part of the fields to return for each result!